iidm/iidm-modification/src/main/java/com/powsybl/iidm/modification/topology/RemoveVoltageLevel.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.modification.topology;
+
+import com.powsybl.commons.PowsyblException;
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.iidm.modification.AbstractNetworkModification;
+import com.powsybl.iidm.network.*;
+import com.powsybl.computation.ComputationManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+import static com.powsybl.iidm.modification.topology.ModificationReports.notFoundVoltageLevelReport;
+
+/**
+ * @author Etienne Homer <etienne.homer at rte-france.com>
+ */
+public class RemoveVoltageLevel extends AbstractNetworkModification {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RemoveVoltageLevel.class);
+
+    private final String voltageLevelId;
+
+    public RemoveVoltageLevel(String voltageLevelId) {
+        this.voltageLevelId = Objects.requireNonNull(voltageLevelId);
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
```

Comment:
To everybody:
- This feature (but we have this issue also for a classical connectable remove) does not handle extensions? Or it is hidden in the remove of each connectable? A `VoltageLevel` can also have extensions. What is our position on that?
- I have an issue with the remove of connectables in general, that we do here. In some network, we are using `regulatingTerminal` that can be remote. So I think that we can remove a connectable which terminal can be used for a remote regulation.

Reply:
About the first point, it is already handled by the `remove` method!

About the second point: it is indeed an issue, to be discussed in another PR about removal in general.